allennlp/training/model.py

```diff
@@ -0,0 +1,94 @@
+from typing import Any, List, Dict, Optional, Union
+import torch
+
+
+class Model(torch.nn.Module):
+    """
+    This abstract class represents a model to be trained. Rather than relying on the
+    raw torch Module, we provide a slightly different API, particularly to align with
+    data processing code for textual data. As textual data pre-processing and deep NLP
+    model inputs tend to be more complicated than for other modalities, we use
+    dictionaries of tensors to pass inputs to the model, to keep on top of what's
+    going on. It also makes reading model implementations easier to connect to
+    data generation, because model inputs are named.
+    """
```

Comment:
I would re-word this comment.  I would emphasize the benefits of this model over the PyTorch model earlier on.  Maybe:

> This abstract class represents a model to be trained.  Rather than relying on the raw torch model, we provide a richer API that maintains more metadata about the tensors used.  This makes reading model implementations easier to connect to data generation, because model inputs are named.

Still, this documentation leaves me wondering whether we should just be using the PyTorch API (I have not looked at the code yet).